### DISCUSSION TITLE
Why is usePrefix overwritten by settings in carbon-components?

### DISCUSSION THREAD

**Discussion by TastyDucks** [carbon-design-system/carbon#9791] [Help] [answered]
### Package

carbon-components, carbon-components-react

### Browser

Chrome, Firefox

### Package version

@carbon/react@0.6.0, carbon-components@10.45.0, carbon-components-react@7.45.0

### Description

Components are not receiving the expected class prefix:
![image](https://user-images.githubusercontent.com/15849445/135644159-cbee3323-2d40-4591-814e-6abda660bbbf.png)

If I'm reading correctly, this may be caused by the prefix set in `carbon-components` [here](https://github.com/carbon-design-system/carbon/blob/90b2b8c089af8fc2f9db25255d4b27d5b16ca5fe/packages/components/src/globals/js/settings.js#L33).

Perhaps the order in which the packages are required causes the prefix in `carbon-components` to override what's done in `react/prefix.js`?

### CodeSandbox example

https://codesandbox.io/s/upbeat-ganguly-k6phd?file=/src/index.js

### Steps to reproduce

I couldn't load `@carbon/react@0.6.0` in the sandbox, but that does illustrate components prefixed with `bx`.
In a fresh repo with `create-react-app`:

`App.scss`:
```scss
@use '@carbon/react';
```

`App.tsx`:
```typescript
import './App.scss' // This loads the cds styles as expected
import { AnyComponent } from '@carbon/react'
export const App = () => <AnyComponent/> // This component will have a bx-prefixed className applied.
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment, marked as answer]
Hey @TastyDucks - sorry for the confusion here! `@carbon/react` is in preview. It's part of the current [beta](https://medium.com/carbondesign/carbon-v11-beta-2-7d0aa4e732d1) for our v11 release and is intended to replace `carbon-components` and `carbon-components-react`.

`@carbon/react` forwards styles from `@carbon/styles` and it also includes the same components (with some v11 updates) from `carbon-components-react`. 

This is outlined in more detail in the [v11 migration guide](https://v11.carbondesignsystem.com/migrating/guide/develop) (also still in beta and a WIP)

So overall, the fix here is to either use `@carbon/react` exclusively for components and styles, or use `carbon-components` with `carbon-components-react`.

I'm not sure what's up with the versioning on codesandbox. I was able to make a fork of your codesandbox with the newer 0.6.0 version of `@carbon/react`:
https://codesandbox.io/s/flamboyant-butterfly-e7n34?file=/package.json

**Reply by TastyDucks** [reply]
Thank you for the quick reply.
To clarify, in the project I am working on I am only using `@carbon/react`.
I made a new `create-react-app` and was able to get classes prefixed with `cds` -- so there must be something wrong with my main build environment. (Removing `node_modules` + `package-lock.json` and reinstalling packages didn't help, which was what led me to believe there may be a bug.)
